### PR TITLE
[FORMS-10944] update filled/empty class

### PR DESCRIPTION
**Pull request by amit-mnnit**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.frontend/src/view/FormFieldBase.js [hunk 1]
```diff
@@ -463,24 +463,22 @@ class FormFieldBase extends FormField {
...
+        const checkedWidget = ['radio', 'checkbox'];
         // radiobutton, checkbox, datefield(AFv1, not datepicker), etc. have multiple widgets in the form of a NodeList
         if (this.widget instanceof NodeList) {
-            this.widget.forEach((widget) => updateModifierClass(widget, (widget.type === "radio" || widget.type === "checkbox") ? widget.checked : widget.value))
+          value = Array.from(this.widget).map((widget) => checkedWidget.includes(widget.type) ? widget.checked : widget.value).find(value => value);
         } else {
-            updateModifierClass(this.widget, this.widget.value)
+          value = checkedWidget.includes(this.widget.type) ? this.widget.checked : this.widget.value;
         }
+        const bemClass = Array.from(this.element.classList).filter(bemClass => !bemClass.includes('--'))[0]
```
Comment (rismehta) [inline ui.frontend/src/view/FormFieldBase.js:477]: Can you add a test case for the existing of this classe based on empty /filled value in runtime ?

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

File: ui.frontend/src/view/FormFieldBase.js [hunk 2]
```diff
@@ -463,24 +463,22 @@ class FormFieldBase extends FormField {
...
-
+        if (!this.getWidget()){
+          return;
+        }
+        let value = '';
+        const checkedWidget = ['radio', 'checkbox'];
         // radiobutton, checkbox, datefield(AFv1, not datepicker), etc. have multiple widgets in the form of a NodeList
         if (this.widget instanceof NodeList) {
-            this.widget.forEach((widget) => updateModifierClass(widget, (widget.type === "radio" || widget.type === "checkbox") ? widget.checked : widget.value))
+          value = Array.from(this.widget).map((widget) => checkedWidget.includes(widget.type) ? widget.checked : widget.value).find(value => value);
```
Comment (sakshi-arora1) [inline ui.frontend/src/view/FormFieldBase.js:473]: what about dropdown?

File: ui.frontend/src/view/FormFieldBase.js (hunk 1 above)
Comment (amit-mnnit) [reply ui.frontend/src/view/FormFieldBase.js:477]: done

File: ui.frontend/src/view/FormFieldBase.js (hunk 2 above)
Comment (amit-mnnit) [reply ui.frontend/src/view/FormFieldBase.js:473]: done

**Review by sakshi-arora1 — APPROVED** [review]

**Review by sakshi-arora1 — APPROVED** [review]

**Review by rismehta — APPROVED** [review]
